comment_limit.module

```diff
@@ -39,76 +39,80 @@ function comment_limit_form_field_config_edit_form_alter(&$form, FormStateInterf
     $form['comment_limit']['user_limit'] = [
       '#type' => 'number',
       '#title' => 'User limit',
-      '#description' => $this->t('The maximum number of comments per user on this entity, e.g. 2 or 0 for no limit'),
+      '#description' => t('The maximum number of comments per user on this entity, e.g. 2 or 0 for no limit'),
       '#weight' => 1,
       '#min' => '0',
       '#default_value' => $default_user,
     ];
-    if ($comment->getThirdPartySetting('comment_limit', 'field_limit', 0) == NULL) {
-      $default_entity = '0';
-    }
-    else {
-      $default_entity = $comment->getThirdPartySetting('comment_limit', 'field_limit', 0);
-    }
+    $default_entity = $comment->getThirdPartySetting('comment_limit', 'field_limit', 0);
     $form['comment_limit']['field_limit'] = [
       '#type' => 'number',
       '#title' => 'Field limit',
-      '#description' => $this->t('The maximum number of comments for this field, e.g. 10 or 0 for no limit'),
+      '#description' => t('The maximum number of comments for this field, e.g. 10 or 0 for no limit'),
       '#weight' => 1,
       '#min' => '0',
       '#default_value' => $default_entity,
     ];
-
     $form['#entity_builders'][] = 'comment_limit_form_field_edit_form_add_form_builder';
   }
 }
 
 /**
- * Implements hook_form_alter().
+ * Implements hook_form_FORM_ID_alter().
  */
-function comment_limit_form_alter(&$form, FormStateInterface $form_state) {
-  if ($form['#form_id'] == 'comment_comment_form') {
-    $comment = $form_state->getFormObject()->getEntity();
-    $comment_limit = Drupal::service('comment_limit.service');
-    $form_action = explode('/', $form['#action']);
-    $last_key = count($form_action) - 1;
-    $field_name = $form_action[$last_key];
-    $entity_id = $comment->getCommentedEntityId();
-    $entity_type = $comment->getCommentedEntityTypeId();
-    $entity = Drupal::entityTypeManager()
-      ->getStorage($entity_type)
-      ->load($entity_id);
-    $entity_bundle = $entity->bundle();
-    $field_config = FieldConfig::loadByName($entity_type, $entity_bundle, $field_name);
-    $field_id = $field_config->id();
-    $field_label = $field_config->getLabel();
-    $definitions = DataDefinition::create('any')
-      ->addConstraint('CommentFormConstraint', [
-        'entityId' => $entity_id,
-        'entityType' => $entity_type,
-        'fieldId' => $field_id,
-        'fieldName' => $field_name,
-      ]);
-    $typed_data = Drupal::typedDataManager()->create($definitions);
-    $violations = $typed_data->validate();
-
-    if ($violations->count() > 0) {
-      $form = [];
-      $form[$field_name] = [
-        '#type' => 'html_tag',
-        '#tag' => 'b',
-        '#value' => $comment_limit->getMessage($entity_id, $entity_type, $field_name, $field_id, $field_label),
-      ];
+function comment_limit_form_comment_form_alter(&$form, FormStateInterface $form_state, $form_id) {
```

Comment:
Have you tried it? The hook_form_FORM_ID_alter() wil not change, just because the id rendered in the markup changes. A {module}_form_comment_form_alter() catches _all_ comment_form forms, whether they are on the same page or not.